src/systems/systems_wasm/src/aspects.rs

```diff
@@ -894,15 +948,23 @@ impl ExecutableVariableAction for OnChange {
         _: &mut SimConnect,
         variables: &mut MsfsVariableRegistry,
     ) -> Result<(), Box<dyn Error>> {
-        let current = variables.read(&self.observed_variable);
+        let current_values: Vec<f64> = variables
+            .read_many(&self.observed_variables)
+            .iter_mut()
+            .map(|v| v.unwrap_or(0.))
+            .collect();
 
-        if let (Some(previous), Some(current)) = (self.previous_value, current) {
-            if previous != current {
-                (self.func)(previous, current);
-            }
+        let has_changed = self
+            .previous_values
+            .iter()
+            .zip(&current_values)
+            .any(|(previous, current)| previous != current);
```

Comment:
Wow :D  Had to open some rust documentation there :D

Reply:
😀